lib/opte/src/engine/ip6.rs

```diff
@@ -632,21 +618,30 @@ pub(crate) mod test {
                     let mut packet =
                         Ipv6HopByHopHeader::new_checked(&mut buf).unwrap();
                     hbh.emit(&mut packet);
-                    hbh.buffer_len()
+                    8 + (hbh.length as usize) * 8
                 }
                 Ipv6Frag => {
                     let frag = fragment_header();
                     let mut packet =
-                        Ipv6FragmentHeader::new_checked(&mut buf).unwrap();
-                    frag.emit(&mut packet);
-                    frag.buffer_len()
+                        Ipv6ExtHeader::new_checked(&mut buf).unwrap();
+                    packet.set_next_header(IpProtocol::Tcp);
+                    let mut frag_packet =
+                        Ipv6FragmentHeader::new_checked(packet.payload_mut())
+                            .unwrap();
+                    frag.emit(&mut frag_packet);
+                    2 + frag.buffer_len()
```

Comment:
that's an annoying change. I guess this is a test, but not a fan of the random magic numbers required now 😕️